### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by korgon**
* adding documentation for patch file usage

### CODE REVIEW

File: docs/patches.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+`[0-9].[0-9].[0-9]`
+
+## Description
+The description describes the changes being made.
+
+## Steps
+These are the steps to take when applying the patch. Any number of steps can be defined and are run sequentially.
+
+### Step type `run`
+Various commands can be executed using the `run` step. In most cases this command would be used to copy over new files, remove existing files or rename files. The commands used here expect a Unix environment of some sort. Commonly utilized commands might include the `cp`, `mv` and `rm` commands.
```
Comment (dkonieczek) [inline docs/patches.md:37]: What do you think about adding a `run` to the example above or here?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,19 @@
 # Snapfu Patches
-This repository contains patch files utilized by Snapfu used to update Snap repositories.
\ No newline at end of file
+This repository contains patch files utilized by Snapfu to update Snap projects. Patch files are ordered by framework and version. Patch versions are automatically created when a new version of Snap is released - this is done within the Snap monorepo repository.
+
```
Comment (dkonieczek) [inline README.md:3]: Should we mentions some requirements here? Ie. only Unix environment support?

File: docs/edit-yaml.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                                steps: [{
+                                    name: New step,
+                                    uses: some/action@v1
+                                }]
+                            }
+                        }
+                    }
+```
+
+Using the `path` method is more verbose, as each path and value must be specified in a separate `update` statement. The verboseness of this method provides more control however, allowing one to specify how to add to existing arrays by providing a `modifier`. The `modifier` currently supports `set`, `append` and `prepend` - `set` is the default if none is provided and will set the property to the provided value (or values), replacing any existing value(s).
```
Comment (dkonieczek) [inline docs/edit-yaml.md:76]: > The verboseness of this method provides more control however, allowing one to specify how to add to existing arrays by providing a `modifier`.

Should 'allowing' be 'allows' here?
OR
provides more control ~~however~~, allowing one

**Review by dkonieczek — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (korgon) [reply README.md:3]: I will add that in the `run` section - as I believe that it is currently the only Unix environment restriction.

File: docs/edit-yaml.md (hunk 3 above)
Comment (korgon) [reply docs/edit-yaml.md:76]: Yea, this does a read a bit awkwardly - I like your second suggestion, I'll edit to something along those lines.

File: README.md (hunk 2 above)
Comment (korgon) [reply README.md:3]: I will also make mention of that in this top level readme.

File: docs/patches.md (hunk 1 above)
Comment (korgon) [reply docs/patches.md]: Example added.

**Review by dkonieczek — APPROVED** [review]

**Pull request merged**